dotnet/src/Microsoft.Agents.AI.Hosting.AzureFunctions/BuiltInFunctions.cs

```diff
@@ -45,55 +46,101 @@ public static async Task<HttpResponseData> RunAgentHttpAsync(
         [DurableClient] DurableTaskClient client,
         FunctionContext context)
     {
-        // The session ID is an optional query string parameter.
-        string? threadIdFromQuery = req.Query["threadId"];
+        // Parse request body - support both JSON and plain text
+        string? message = null;
+        string? threadIdFromBody = null;
+
+        if (req.Headers.TryGetValues("Content-Type", out IEnumerable<string>? contentTypeValues) &&
+            contentTypeValues.Any(ct => ct.Contains("application/json", StringComparison.OrdinalIgnoreCase)))
+        {
+            // Parse JSON body using POCO record
+            AgentRunRequest? requestBody = await req.ReadFromJsonAsync<AgentRunRequest>(context.CancellationToken);
+            if (requestBody != null)
+            {
+                message = requestBody.Message;
+                threadIdFromBody = requestBody.ThreadId;
+            }
+        }
+        else
+        {
+            // Plain text body
+            message = await req.ReadAsStringAsync();
+        }
+
+        // The thread ID can come from query string or JSON body
```

Comment:
NIT: Want to add support for header `x-ms-thread-id` too?

Reply:
This would mean there are three different ways to specify the thread ID (JSON body, query string parameter, and header), which complicates request validation. If we support these, we need to come up with a set of rules for how to validate these requests to ensure consistency or decide which thread ID should be the "winner". I'd like to defer that for now since it will make the code logic more complicated and I don't want to spend time writing more tests to cover these other use cases.